sensors/triggers/kafka/kafka.go

```diff
@@ -63,6 +63,24 @@ func NewKafkaTrigger(sensor *v1alpha1.Sensor, trigger *v1alpha1.Trigger, kafkaPr
 			config.Version = version
 		}
 
+		if kafkatrigger.SASL != nil {
+			config.Net.SASL.Enable = true
+			config.Net.SASL.Mechanism = sarama.SASLMechanism(kafkatrigger.SASL.Mechanism)
```

Comment:
Validating?

Reply:
Checks whether the configuration is set. Enables SASL authentication if parameters are set in the YAML configs.

Reply:
Added additional validation for empty strings.